interop/lib/interop/client.ex

```diff
@@ -169,6 +161,25 @@ defmodule Interop.Client do
     [] = Enum.to_list(res_enum)
   end
 
+  @doc """
+  We build the Stream struct (res_enum) using Stream.unfold/2
+  the unfold function is built in such a way - for both adapters - that the acc is map used to find a
+  connection_stream process and the next_fun arg is a function that reads directly from the connection_stream
+  that is producing data.
+  Every time we execute the next_fun we read a chunk of data and remove that from the buffer.
+  That action by itself will generate a side effect that will update the state of the connection_stream by removing the chunk of data we just read.
+  An easier way to visualize that is. Take the code and the execution bellow as an example
+
+
+  ```
+  iex(4)> ex_stream |> Stream.take(1) |> Enum.to_list()
+  [1]
+  iex(5)> ex_stream |> Enum.to_list()
+  [2, 3]
+  iex(6)> ex_stream |> Enum.to_list()
+  []
+  ```
```

Comment:
```suggestion
  ```

  ## Examples

      iex> ex_stream |> Stream.take(1) |> Enum.to_list()
      [1]
      iex> ex_stream |> Enum.to_list()
      [2, 3]
      iex> ex_stream |> Enum.to_list()
      []
  ```
```

Reply:
By the way, this example actually got me thinking if there's a way to make the stream read-only by default and have it become stateful only by doing something more explicit to it.

Reply:
For both adapters yes, tho, I would need to decouple the connection stream from the unfold function. That by itself would require huge refactor :/ We could add address that in a incoming PR

Reply:
If we haven't already, let's leave a to-do and resolve this convo :)